modules/fuse/adaptors/arrow_ipc/serializer_registry.cc

```diff
@@ -89,24 +89,29 @@ std::shared_ptr<arrow::Buffer> arrow_view(
 
 static void from_arrow_view(Client* client, std::string const& path,
                             arrow::io::RandomAccessFile* fp) {
-  std::shared_ptr<arrow::ipc::RecordBatchFileReader> reader;
+  std::shared_ptr<arrow::ipc::RecordBatchStreamReader> reader;
   CHECK_ARROW_ERROR_AND_ASSIGN(reader,
-                               arrow::ipc::RecordBatchFileReader::Open(fp));
+                               arrow::ipc::RecordBatchStreamReader::Open(fp));
 
-  std::vector<std::shared_ptr<arrow::RecordBatch>> batches;
-  for (int64_t index = 0; index < reader->num_record_batches(); ++index) {
-    std::shared_ptr<arrow::RecordBatch> batch;
-    CHECK_ARROW_ERROR_AND_ASSIGN(batch, reader->ReadRecordBatch(index));
-    batches.emplace_back(batch);
-  }
+  // std::vector<std::shared_ptr<arrow::RecordBatch>> batches;
+  // for (int64_t index = 0; index < reader->(); ++index) {
+  //   std::shared_ptr<arrow::RecordBatch> batch;
+  //   CHECK_ARROW_ERROR_AND_ASSIGN(batch, reader->ReadRecordBatch(index));
+  //   batches.emplace_back(batch);
+  // }
 
   std::shared_ptr<arrow::Table> table;
+  std::vector<std::shared_ptr<arrow::RecordBatch>> batches;
+
+  VINEYARD_CHECK_OK(reader->ReadAll(&batches));
+
   VINEYARD_CHECK_OK(RecordBatchesToTable(batches, &table));
 
   // build it into vineyard
   TableBuilder builder(*client, table);
   auto tb = builder.Seal(*client);
-  VINEYARD_CHECK_OK(client->Persist(tb->id()));
+  // VINEYARD_CHECK_OK(client->Persist(tb->id()));
```

Comment:
Delete this line if you didn't need to check if has been persisted.

Reply:
I am not sure, since this commit does not support the write, I leave it here for future development, I want to discuss this design with you I will open an another discussion thread about the implementation of fuse write tomorrow : >